flowmachine/flowmachine/core/cache.py

```diff
@@ -0,0 +1,384 @@
+    """
+    Get the upper limit set in FlowDB for the cache size, in bytes.
+
+    Parameters
+    ----------
+    connection : "Connection"
+
+    Returns
+    -------
+    int
+        Number of bytes in total available cache tables
+
+    """
+    sql = "SELECT cache_max_size()"
+    return int(connection.fetch(sql)[0][0])
+
+
+def get_cache_half_life(connection: "Connection") -> float:
+    """
+    Get the current setting for cache half-life.
+
+    Parameters
+    ----------
+    connection : "Connection"
+
+    Returns
+    -------
+    float
+        Cache half-life setting
+
+    """
+    sql = "SELECT cache_half_life()"
+    return float(connection.fetch(sql)[0][0])
+
+
+def set_cache_half_life(connection: "Connection", cache_half_life: float) -> None:
+    """
+    Set the cache's half-life.
+
+    Parameters
+    ----------
+    connection : "Connection"
+    cache_half_life : float
+        Setting for half-life
+
+    Notes
+    -----
+
+    Changing this setting without flushing the cache first may have unpredictable consequences
+    and should be avoided.
+    """
+    sql = (
+        f"UPDATE cache.cache_config SET value='{cache_half_life}' WHERE key='half_life'"
+    )
+    with connection.engine.begin() as trans:
+        trans.execute(sql)
+
+
+def set_max_size_of_cache(connection: "Connection", cache_size_limit: int) -> None:
+    """
+    Set the upper limit set in FlowDB for the cache size, in bytes.
+
+    Parameters
+    ----------
+    connection : "Connection"
+    cache_size_limit : int
+        Size in bytes to set as the cache limit
+
+    """
+    sql = f"UPDATE cache.cache_config SET value='{cache_size_limit}' WHERE key='cache_size'"
+    with connection.engine.begin() as trans:
+        trans.execute(sql)
+
+
+def compute_time(connection: "Connection", query_id: str) -> float:
+    """
+    Get the time in ms that a cached query took to compute.
+
+    Parameters
+    ----------
+    connection : "Connection"
+    query_id : str
+        md5 identifier of the query
+
+    Returns
+    -------
+    float
+        Number of seconds the query took to compute
+
+    """
+    try:
+        return float(
+            connection.fetch(
+                f"SELECT compute_time FROM cache.cached WHERE query_id='{query_id}'"
+            )[0][0]
+            / 1000
+        )
+    except IndexError:
+        raise ValueError(f"Query id '{query_id}' is not in cache on this connection.")
+
+
+def score(connection: "Connection", query_id: str) -> float:
+    """
+    Get the current cache score for a cached query.
+
+    Parameters
+    ----------
+    connection: "Connection"
+    query_id : str
+        md5 id of the cached query
+
+    Returns
+    -------
+    float
+        Current cache score of this query
+
+    """
+    try:
+        return float(
+            connection.fetch(
+                f"""SELECT cache_score_multiplier*((compute_time/1000)/table_size(tablename, schema))
```

Comment:
This formula is repeated in a couple of places. Worth wrapping it up in a helper function with a meaningful name?